api/geth_backend.go

```diff
@@ -619,27 +618,31 @@ func (b *GethStatusBackend) ConvertToKeycardAccount(account multiaccounts.Accoun
 	if err != nil {
 		return err
 	}
-	err = accountDB.SaveSettingField(settings.KeycardInstanceUID, s.KeycardInstanceUID)
+
+	keypair, err := accountDB.GetKeypairByKeyUID(account.KeyUID)
 	if err != nil {
+		if err == accounts.ErrDbKeypairNotFound {
+			return errors.New("cannot convert an unknown keypair")
+		}
 		return err
 	}
 
-	err = accountDB.SaveSettingField(settings.KeycardPairedOn, s.KeycardPairedOn)
+	err = accountDB.SaveSettingField(settings.KeycardInstanceUID, s.KeycardInstanceUID)
 	if err != nil {
 		return err
 	}
 
-	err = accountDB.SaveSettingField(settings.KeycardPairing, s.KeycardPairing)
+	err = accountDB.SaveSettingField(settings.KeycardPairedOn, s.KeycardPairedOn)
```

Comment:
I haven't touched this, don't know why it's rendered like this by github diff. The only change I did here is that I added this line `keypair, err := accountDB.GetKeypairByKeyUID(account.KeyUID)` instead of the old line `relatedAccounts, err := accountDB.GetAccountsByKeyUID(account.KeyUID)`. The rest remained as it was before.

Reply:
Explained above.